eos_data_distribution/avahi_monitor.py

```diff
@@ -79,23 +96,102 @@ def __init__(self):
         check_call(
             ["nfdc", "set-strategy", str(SUBSCRIPTIONS_SOMA), 'ndn:/localhost/nfd/strategy/multicast'])
 
-        self._registry = dict()
+        self.update_routed_status()
+        self.process_route_changes(False)
+
+    def update_routed_status(self):
+        self._routed = set(
+            self._ndn_gateways.values()).intersection(self.gateways)
+
+    def process_route_changes(self, prev_routed):
+        if self._routed:
+            logger.info(
+                "Being routed by an NDN node at %s (via %s), NOT acting as an edge router", address, ifname)
+            if not prev_routed:
+                # need to remove old links...
+                self._registery.values().map(self.remove_nexthop)
```

Comment:
s/`self._registery`/`self._registry`/ (which suggests that this code path hasn’t been tested?)